Sources/TuistCoreTesting/Utils/StubClock.swift

```diff
@@ -0,0 +1,48 @@
+import Foundation
+import TuistCore
+
+/// A stub clock that can be primed with
+/// multiple dates and timers.
+public class StubClock: Clock {
+    /// Returns primed dates in the order they
+    /// were added to `primedDates`.
+    ///
+    /// In the event there are no primed dates, `Date.distantPast`
+    /// is returned.
+    public var now: Date {
+        if let first = primedDates.first {
+            primedDates.removeFirst()
+            return first
+        }
+        return .distantPast
+    }
+
+    public var primedDates: [Date] = []
+    public var primedTimers: [TimeInterval] = []
+
+    public init() {}
+
+    /// Returns stub timers that are primed with time intervals in the
+    /// the order they were defined in `primedTimers`.
+    ///
+    /// In the event there are no primed time intervals, A stub timer
+    /// with a time interval of `0` is returned.
+    public func startTimer() -> ClockTimer {
+        if let first = primedTimers.first {
+            primedTimers.removeFirst()
+            return Timer(timeInterval: first)
+        }
+        return Timer(timeInterval: 0.0)
```

Comment:
What about XCTAsserting an error here if the test subject calls `stop` more times than the number of `primedTimers` that we've set? We can print an error like:

```
Trying to get more timers than the ones stubbed
```

That'll help debugging the test in the future.

Reply:
Good idea 💡

while experimenting with it, I found it hindered testing slightly. e.g. for the other test cases that aren't testing the timer, it would be a shame to burden them with priming times / dates for this stub. As such I added an option to enable/disable this assert.